thewalrus/internal_modes/prepare_cov.py

```diff
@@ -0,0 +1,304 @@
+
+    Args:
+        F (list[list/array[array]): List of arrays of the temporal modes of the Schmidt modes in each spatial mode, must be normalized and ordered by spatial mode.
+                                    Sufficient to put the output temporal modes from yellowsubmaring in a list ordered by spatial mode.
+    Returns:
+        (array): the overlap matrix
+    """
+    Fs = [f for fj in F for f in fj]
+    R = len(Fs)
+    O = np.eye(R, dtype=np.complex128)
+    for j in range(R):
+        for k in range(R):
+            O[j, k] = np.inner(Fs[j].conj() / np.linalg.norm(Fs[j]), Fs[k] / np.linalg.norm(Fs[k]))
+    return O
+
+
+def vacuum_state(M, hbar=2):
+    r"""
+    Returns the displacement vector and covariance matrix for an M mode vacuum state.
+
+    Args:
+        M (int): number of modes (spatial, Schmidt, other etc.)
+        hbar (float): (default 2) the value of :math:`\hbar` in the commutation relation :math:`[\x,\p]=i\hbar`
+
+    Returns:
+        tuple[array]: the displacement vector and covariance matrix of an M mode vacuum state
+    """
+
+    return np.zeros(2 * M), np.eye(2 * M) * hbar / 2
+
+
+# pylint: disable=too-many-branches
+def orthonormal_basis(rjs, O=None, F=None, thr=1e-3):
+    r"""
+    Finds an orthonormal basis in operator space based on the total number of Schmidt modes in the whole system of multiple spatial modes.
+    Using the overlap matrix and the squeezing parameters of each Schmidt mode in each waveguide it finds R orthonormal modes for each spatial mode,
+    where R is less than or equal to the total combined number of Schmidt modes in the whole system of multiple spatial modes.
+
+    Args:
+        rjs (list[list]): list for each spatial mode of list of squeezing parameters in that spatial mode
+        O (array): 2-dimensional matrix of the overlaps between each Schmidt mode in all spatial modes combined
+        F (list[list/array[array]): List of arrays of the temporal modes of the Schmidt modes in each spatial mode, must be normalized and ordered by spatial mode.
+                                    Sufficient to put the output temporal modes from yellowsubmaring in a list ordered by spatial mode.
+        thr (float): eigenvalue threshold under which orthonormal mode is discounted
+
+    Returns:
+        tuple (chis, eps, W): chis is a list of the temporal functions for the new orthonormal basis, only returned when F is given
+                              eps is a list of arrays, in each array is the effective squeezing parameters for the orthonormal modes in each spatial mode
+                              W is a list of arrays, each array is an R x R matrix for each spatial mode for the amplitude of each orthonormal mode.
+    """
+    rs = np.array([r for rj in rjs for r in rj])
+    if F is not None:
+        Fs = [f for fj in F for f in fj]
+        Fs = [Fs[j] / np.linalg.norm(Fs[j]) for j in range(len(Fs))]
+        if not np.allclose(len(Fs), rs.shape[0]):
```

Comment:
```suggestion
        if len(Fs) != rs.shape[0]:
```
Why not just compare the two integers exactly?

Reply:
It would be absolutely fine in this case, but I have an aversion to using anything other than np.allclose for comparisons in general